### DISCUSSION TITLE
Configuring Flight Gear with JSBsim

### DISCUSSION THREAD

**Discussion by adeelahsaanawan** [JSBSim-Team/jsbsim#895] [Q&A]
Hi everyone!

I am learning how to configure JSBsim with Flight gear and I could not find any documentation. Can anyone please guide me or refer me to some documentation or video where I can learn how to configure JSBsim with Flightgear step by step. I really appreciate the help.

**Comment by seanmcleod** [comment]
What sort of configuration are you thinking of, or think you need? JSBSim is natively supported by FlightGear as a FDM (Flight Dynamics Model) engine, so if you simply load up one of the aircraft included with FlightGear that are JSBSim based models, or download one of hundreds available on the internet then you can simply fly the aircraft model in FlightGear, with JSBSim acting as the FDM engine.

So there is nothing really specifically that you need to configure. So you'll have to describe in more detail what you're wanting to do.

**Comment by adeelahsaanawan** [comment]
I am quite new to this, so excuse me if I ask something silly.  I am trying to understand how to use the UDP protocol etc. for JSBSim and FlightGear. There is a way to do it where you set the port and JSBSim when run should output data on that port and flightgear will read data on that port. I could not find any documentation on that. Also, how do you know which airplanes are based on JSBSim?

**Comment by seanmcleod** [comment]
Okay, so if you're wanting to use the UDP FlightGear protocol then you're saying you want to run JSBSim stand-alone in a separate process and have it output FlightGear UDP packets to drive FlightGear externally on the same PC or a PC on the network, as opposed to simply opening up the aircraft within FlightGear and running it from within FlightGear, in which case there is no need for configuring the UDP FlightGear output etc.

There may be easier ways to tell from within FlightGear, but otherwise take a look at the location of the aircraft files and you can tell by opening the aircraft.xml file and see if the XML looks similiar to this example:

https://github.com/JSBSim-Team/jsbsim/blob/master/aircraft/737/737.xml

**Comment by adeelahsaanawan** [comment]
Yes. For this part "Okay, so if you're wanting to use the UDP FlightGear protocol then you're saying you want to run JSBSim stand-alone in a separate process and have it output FlightGear UDP packets to drive FlightGear externally on the same PC or a PC on the network"

Is there any documentation available or any relative webpages?

**Comment by seanmcleod** [comment]
Take a look at - https://github.com/JSBSim-Team/jsbsim/blob/375f5be0fdfeb210f4cf66f938e3e6fd426c2966/data_output/flightgear.xml

**Comment by adeelahsaanawan** [comment]
alright, Thanks!
